### PR TITLE
CI: Rewrite for both OBS 27 and 28

### PR DESCRIPTION
**Pull request by norihiro**
Major objectives are
- Provides packages for OBS Studio 27 and OBS Studio 28
- Small enough plugin-specific CI and cmake code.
- `cmake .. && make && make install` flow should work once a user correctly configure their environment. (necessary for my development, also easier for other packagers such as AUR.)

See also
- norihiro/obs-color-monitor/pull/56

Known issue(s)
- [ ] openblas is not installed for macOS arm64. This is because `brew install` did not install the library for arm64.

- [x] The commit is reviewed by yourself.
- [x] The code is tested.
- [x] Document is up to date or not necessary to be changed.
- [x] The commit is compatible with repository's license.

### CODE REVIEW

**Comment by ogmkp** [comment]
Why a rewrite for OBS 27 ? It was not compatible ?

**Comment by norihiro** [comment]
> Why a rewrite for OBS 27 ? It was not compatible ?

Old code is working for OBS 27.2. However I don't want to have different CI scripts and cmake files for both 27 and 28.

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,165 +78,48 @@ set(PLUGIN_SOURCES
...
 
-# -- OS X specific build settings and tasks --
 if(APPLE)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++ -fvisibility=default")
 
 	set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PREFIX "")
-	target_link_libraries(${CMAKE_PROJECT_NAME} "${OBS_FRONTEND_LIB}")
+	set(MACOSX_PLUGIN_GUI_IDENTIFIER "${MACOS_BUNDLEID}")
+	set(MACOSX_PLUGIN_BUNDLE_VERSION "${CMAKE_PROJECT_VERSION}")
+	set(MACOSX_PLUGIN_SHORT_VERSION_STRING "1")
```
Comment (norihiro) [inline CMakeLists.txt:120]: Is this necessary?
If so, the code should be 0.

File: cmake/ObsPluginHelpers.cmake [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+				FATAL_ERROR
+				"No 'MACOSX_PLUGIN_SHORT_VERSION_STRING' set, but is required to build plugin bundles on macOS - example: '1.0.2'"
+				)
+		endif()
+
+		set(MACOSX_PLUGIN_BUNDLE_NAME "${target}" PARENT_SCOPE)
+		set(MACOSX_PLUGIN_BUNDLE_VERSION "${MACOSX_BUNDLE_BUNDLE_VERSION}" PARENT_SCOPE)
+		set(MACOSX_PLUGIN_SHORT_VERSION_STRING "${MACOSX_BUNDLE_SHORT_VERSION_STRING}" PARENT_SCOPE)
+		set(MACOSX_PLUGIN_EXECUTABLE_NAME "${target}" PARENT_SCOPE)
+		set(MACOSX_PLUGIN_BUNDLE_TYPE "BNDL" PARENT_SCOPE)
```
Comment (norihiro) [inline cmake/ObsPluginHelpers.cmake:138]: Are these variable necessary?

**Pull request merged**